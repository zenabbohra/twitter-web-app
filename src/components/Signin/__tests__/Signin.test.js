import React from 'react';
import {shallow} from 'enzyme';
import SignIn from "../Signin";

describe('<Signin />', () => {
  describe('renders', () => {
    const subject = shallow(<SignIn/>);

    it('has two <TextField/>', () => {
      const textFields = subject.find('WithStyles(ForwardRef(TextField))');
      expect(textFields.length).toBe(2);
      expect(textFields).toMatchSnapshot();
    });

    it('has one <Button/>', () => {
      const button = subject.find('WithStyles(ForwardRef(Button))');
      expect(button.length).toBe(1);
      expect(button).toMatchSnapshot();
    });

    it('has one <Link/>', () => {
      const link = subject.find('WithStyles(ForwardRef(Link))');
      expect(link.length).toBe(1);
      expect(link).toMatchSnapshot();
    });
  });
});