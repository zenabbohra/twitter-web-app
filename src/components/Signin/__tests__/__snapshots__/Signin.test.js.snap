// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin /> renders has one <Button/> 1`] = `
<WithStyles(ForwardRef(Button))
  className="submit"
  color="primary"
  fullWidth={true}
  onClick={[Function]}
  type="submit"
  variant="contained"
>
  Sign In
</WithStyles(ForwardRef(Button))>
`;

exports[`<Signin /> renders has one <Link/> 1`] = `
<WithStyles(ForwardRef(Link))
  href="#"
  onClick={[Function]}
  variant="body2"
>
  Don't have an account? Sign Up
</WithStyles(ForwardRef(Link))>
`;

exports[`<Signin /> renders has two <TextField/> 1`] = `
Array [
  <WithStyles(ForwardRef(TextField))
    autoComplete="email"
    autoFocus={true}
    fullWidth={true}
    id="email"
    label="Email Address"
    margin="normal"
    name="email"
    onChange={[Function]}
    required={true}
    variant="outlined"
  />,
  <WithStyles(ForwardRef(TextField))
    autoComplete="current-password"
    fullWidth={true}
    id="password"
    label="Password"
    margin="normal"
    name="password"
    onChange={[Function]}
    required={true}
    type="password"
    variant="outlined"
  />,
]
`;
